### DISCUSSION TITLE
updateOrCreateMany

### DISCUSSION THREAD

**Discussion by iammrsea** [adonisjs/core#1524] [Help] [answered]
I tried using ```updateOrCreateMany``` model method, but ended up with this error message ```Cannot define "transactionListener" on "Product" model, since it is not defined as a model property```.
Please, does anyone have any ideas how this can be fixed?

**Comment by thetutlage** [comment]
Impossible to help without looking at any code.

**Reply by iammrsea** [reply]
> Impossible to help without looking at any code.

Below is a snippet of my code
```javascript
const products = [
{id:1,price:4000, qty: 10},
{id:3,price:5000, qty: 6},
]
await Product.updateOrCreateMany("id", products);
```

**Reply by KWINGLION** [reply]
can you  post the whole model code and the error here?

**Reply by iammrsea** [reply]
> can you post the whole model code and the error here?

```javascript

 await payment.preload("order", (order) => order.preload("products"));
    payment.verified = true;
    payment.customerCode = data.customer.customer_code;
    payment.customerAuthorizationCode = data.authorization.authorization_code;
    payment.order.status = "completed";
    payment.subaccountCode = data.subaccount && data.subaccount.subaccount_code;
    payment.channel = data.channel;
    await payment.save();
    await payment.order.save();
    const products = payment.order.products.map((product) => {
      product.availableQty = product.availableQty - product.qty_bought;
      return product;
    });
    await Product.updateOrCreateMany("id", products);
```
Error message:
```Cannot define "transactionListener" on "Product" model, since it is not defined as a model property```

**Comment by Reptiluka** [comment, marked as answer]
@iammrsea I got the same problem today and solved it !
I think in your code you return the object/Model Product, and not the serialized version like you think it was in your post.
When I have an array of object, I do this : 
`await Event.fetchOrCreateMany('event_id', arrayEvent.map(ev => ev.toJSON()))`

So in your code, I think you should write : 
```
const products = payment.order.products.map((product) => {
      product.availableQty = product.availableQty - product.qty_bought;
      return product;
    });
    await Product.updateOrCreateMany("id", products.map(prod => prod.toJSON()));
```
OR 
```
    const serialized_products = payment.order.products.map((product) => {
      product.availableQty = product.availableQty - product.qty_bought;
      return product.toJSON();
    });
    await Product.updateOrCreateMany("id", products);
```

**Feature Request** @thetutlage  : Accept model in .updateOrCreateMany and other functions 😀

**Reply by iammrsea** [reply]
Thanks man, that was very helpful
